### DISCUSSION TITLE
Getting pixel data drawn with Painter

### DISCUSSION THREAD

**Discussion by nordsoyv** [emilk/egui#1416] [Q&A] [answered]
Hi. I'm creating a simple app where the user can manipulate some settings then I render some graphics using Painter.
I would like to be able for the user to save this to an image. So is there any way to get access to the rendered pixel data that was created, so that I can use it to create an image and save it to disk ? 

Or is there perhaps an easier way to do this?

**Comment by emilk** [comment, marked as answer]
All current egui backends renders using the GPU, so you will need to copy the region from GPU, which is certainly possible with e.g. `glReadPixels`. There is also some work on a software renderer: https://github.com/emilk/egui/issues/1129

**Reply by nordsoyv** [reply]
Thanks. So since I'm using eframe, I can probably get access to the gl context through the Context object and then use glReadPixels from there . I'll dig around a bit , and see what I can find.

**Reply by emilk** [reply]
You need to access the `glow::Context` _after_ egui does the painting (which after `App::update` is done running), and `eframe` doesn't currently expose that. It is however something I'd like to expose. Somehow. I may add it to the `eframe::Frame` so you can access it on the next call to `App::update`. If you store the `Rect` where you painted your widget, you can then use `gl.read_pixels` to access it.

**Reply by emilk** [reply]
Here is a PR to expose what you need: https://github.com/emilk/egui/pull/1425

**Reply by nordsoyv** [reply]
Thanks a lot
